test/utils/HookMiner.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.21;
```

Comment:
any specific reason we use different versions (0.8.19, 0.8.21, 0.8.24) in different files?

Reply:
i copied HookMiner from another repo so it was 0.8.21 - no reason for the others. i can update them all to 0.8.24